visualization/app/codeCharta/ui/copyToClipboardButton/copyToClipboard.service.ts

```diff
@@ -0,0 +1,22 @@
+import { Inject, Injectable } from "@angular/core"
+import { State } from "../../state/angular-redux/state"
+import { accumulatedDataSelector } from "../../state/selectors/accumulatedData/accumulatedData.selector"
+import { buildTextOfFiles } from "./util/clipboardString"
+import { getFilenamesWithHighestMetrics } from "./util/getFilenamesWithHighestMetrics"
+
+@Injectable()
+export class CopyToClipboardService {
+	constructor(@Inject(State) private state: State) {}
+
+	private getUnifiedMapNode() {
+		const { unifiedMapNode } = accumulatedDataSelector(this.state.getValue())
+		return unifiedMapNode
+	}
+
+	getClipboardText(): string {
+		const node = this.getUnifiedMapNode()
+		const filesByAttribute = getFilenamesWithHighestMetrics(node)
+
+		return buildTextOfFiles(filesByAttribute)
+	}
```

Comment:
I do not think that your brain is small! xD

Personally, I prefer it this way. `getClipboardText` becomes a table of content and the function instantly reveals its structure and intentions:

1. Get a node (unified map node)
2. Get file names by attribute (highest metrics)
3. Build text
4. Return text

So in the end, there is less boiler code in `getClipboardText`, because we get every variable with one function call.
And whoever wants to analyze _how_ we get `unifiedMapNode` can dive into `getUnifiedMapNode`. There, you find the details and nothing but the details.